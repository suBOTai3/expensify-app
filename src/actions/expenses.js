import { v4 as uuid } from 'uuid';

export const addExpense = ({
    amount = 0,
    createdAt = 0,
    description = '',
    note = '' } = {}) => ({
        type: 'ADD_EXPENSE',
        expense: {
            id: uuid(),
            description,
            note,
            amount,
            createdAt
        }
    })

export const removeExpense = ({ id } = {}) =>
({
    type: 'REMOVE_EXPENSE',
    id    
})

export const editExpense = (id, updates) =>
({
    type: 'EDIT_EXPENSE',
    id,
    updates
})
