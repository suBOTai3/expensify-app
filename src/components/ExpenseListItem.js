import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({description, amount, id, createdAt}) =>  (
    <div>
        <Link to={`/edit/${id}`} >
            <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount).format('$R0,000.00')}
            -
            {moment(createdAt).format('yyyy-MMM-DD')}
        </p>        
    </div>
);
 
export default ExpenseListItem;
    