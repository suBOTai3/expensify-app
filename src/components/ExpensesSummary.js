import React from 'react';
import numeral from 'numeral';
import getExpenseTotals from '../selectors/expenses-total'

export class ExpensesSummary extends React.Component {
    constructor(props) {
        super(props)
    }

    getExpenseTotal = (expenses) => {
        getExpenseTotals(expenses);
    }

    render() {
        return (
            <div>
                <p>Viewing {this.props.expenses.length} expense {this.props.expenses.length > 0 && 's'} totalling {numeral(getExpenseTotals(expenses))}</p>
            </div>
        )
    }
}