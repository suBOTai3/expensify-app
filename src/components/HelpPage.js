import React from 'react';
const HelpPage = () => (
    <div>
        From HelpPage component
    </div>
)


export default HelpPage;