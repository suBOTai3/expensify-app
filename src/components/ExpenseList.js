import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                    props.expenses.map((expense) =>
                        <ExpenseListItem key={expense.id} {...expense} dispatch={props.dispatch} />
                    )

                )
        }
    </div>
);

const mapStateToProps = (state) => {
    console.log('mapping state in ExpenseList', state);
    return {
        expenses: selectExpenses(state.expenses, state.filters),
        dispatch: state.dispatch,
    };
};

export default connect(mapStateToProps)(ExpenseList);
