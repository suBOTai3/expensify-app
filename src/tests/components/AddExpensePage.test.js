import React from 'react';
import { shallow } from 'enzyme';
import { AddExpensePage } from '../../components/AddExpensePage';
import expenses from '../fixtures/expenses';
    

let onSubmitSpy, historySpy, wrapper;

beforeEach(() => {
    onSubmitSpy = jest.fn();
    historySpy = { push: jest.fn() };
    wrapper = shallow(<AddExpensePage addExpense={onSubmitSpy} history={historySpy} />, { disableLifecycleMethods: true });
});

test('should render add expense page', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(onSubmitSpy).toHaveBeenLastCalledWith(expenses[0]);
    expect(historySpy.push).toHaveBeenLastCalledWith('/');
})

