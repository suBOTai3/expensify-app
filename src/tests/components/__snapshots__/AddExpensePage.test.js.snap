// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render add expense page 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
  From expense component
</div>
`;
